portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -21,33 +21,66 @@
 import javax.servlet.http.HttpServletResponse;
 import com.google.gson.Gson;
 import java.util.ArrayList;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  /**
-   * Premade list of comments to pass from server
-   */
-  private final ArrayList<String> comments = new ArrayList<>();
   private final Gson gson = new Gson();
-
-  /**
-   * init() is called once when servlet is created
-   */
-  @Override
-  public void init() {
-      comments.add("Pineapple on pizza is great");
-      comments.add("Dogs are overrated");
-      comments.add("Alpacas are awesome");
-  }
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
   /**
    * Get the list of comments from server as a Json
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+
+    int maxComms = Integer.parseInt(request.getParameter("maxComms"));
+
+    // Create a new array, otherwise there are duplicate comments
+    ArrayList<String> comments = new ArrayList<>();
+    for(Entity entity : results.asIterable()) {
+      if(comments.size() == maxComms) {
+          break;
+      }
+
+      String comment = (String) entity.getProperty("message");
+      comments.add(comment);
+    }
+
     String json = gson.toJson(comments);
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
+
+  /**
+   * Let the user post his comment to the server
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment = getUserComment(request, "comment-input", "");
+    long timestamp = System.currentTimeMillis();
+
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("message", comment);
+    commentEntity.setProperty("timestamp", timestamp);
+    datastore.put(commentEntity);
+
+    response.sendRedirect("/index.html");
+  }
+
+  private String getUserComment(HttpServletRequest request, String commentForm, String defaultValue) {
+    String comment = request.getParameter(commentForm);
+    if(comment == null) {
+      return defaultValue;
+    }
+    return comment;
```

Comment:
you can return a ternary here: `return comment == null ? defaultValue : comment`

Reply:
Aren't ternaries more confusing? I thought ifs are preferred.

Reply:
when it is a simple condition like this one, using a ternary is a little less verbose and still easily readable